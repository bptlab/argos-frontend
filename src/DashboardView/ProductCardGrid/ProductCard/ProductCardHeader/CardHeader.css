.ProductCardTitle {
  width: 100%;
  text-align: center;
  padding: 0.6vw 0 0.5vw 0;
  border-bottom: 1px solid white;
  font-size: 1.6vh;
  color: white; }

.ProductCardTitle_RUNNING {
  background: #B1BE6C !important; }

.ProductCardTitle_WARNING {
  background: #F0A57B !important; }

.ProductCardTitle_ERROR {
  background: #DF737A !important; }
