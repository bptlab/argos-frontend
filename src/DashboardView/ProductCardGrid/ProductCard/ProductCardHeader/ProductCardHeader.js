import React, { Component } from 'react';

class ProductCardHeader extends Component {
    render() {
        return (
            <h4 className={ `card-title ProductCardTitle ProductCardTitle_${this.props.category}` }>
                {this.props.children}
            </h4>
        );
    }
}

export default ProductCardHeader;