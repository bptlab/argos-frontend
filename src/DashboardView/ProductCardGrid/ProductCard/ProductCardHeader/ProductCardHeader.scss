@import '../../../../definitions.scss';
.ProductCardTitle {
    width: 100%;
    text-align: center;
    padding: 0.6vw 0 0.5vw 0;
    border-bottom: 1px solid white;
    font-size: 1.6em;
    color: white;
    @media (max-width: ) {

    }
}

.ProductCardTitle_RUNNING {
    background: $COLOR-success-light !important;
}

.ProductCardTitle_WARNING {
    background: $COLOR-warning-light !important;
}

.ProductCardTitle_ERROR {
    background: $COLOR-error-light !important;
}