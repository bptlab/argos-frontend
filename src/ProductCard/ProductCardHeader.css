@font-face {
  font-family: 'Glyphter';
  src: url("../assets/fonts/icons/Glyphter.eot");
  src: url("../assets/fonts/icons/Glyphter.eot?#iefix") format("embedded-opentype"), url("../assets/fonts/icons/Glyphter.woff") format("woff"), url("../assets/fonts/icons/Glyphter.ttf") format("truetype"), url("../assets/fonts/icons/Glyphter.svg#Glyphter") format("svg");
  font-weight: normal;
  font-style: normal; }

.ProductCardTitle {
  width: 100%;
  text-align: center;
  padding: 0.5vw 2vw;
  border-bottom: 1px solid white;
  font-size: 1.6vh; }

.ProductCardTitle_success {
  background: #689F38 !important; }

.ProductCardTitle_warning {
  background: #FFA000 !important; }

.ProductCardTitle_error {
  background: #E64A19 !important; }
