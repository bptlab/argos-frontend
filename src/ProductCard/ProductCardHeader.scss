@import '../App/App.scss';
.ProductCardTitle {
    width: 100%;
    text-align: center;
    padding: 0.5vw 2vw;
    border-bottom: 1px solid white;
    font-size: 1.6vh;
}

.ProductCardTitle_success {
    background: $COLOR-success-light !important;
}

.ProductCardTitle_warning {
    background: $COLOR-warning-light !important;
}

.ProductCardTitle_error {
    background: $COLOR-error-light !important;
}