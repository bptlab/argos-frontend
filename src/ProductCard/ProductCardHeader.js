import React, { Component } from 'react';
import {CardTitle} from 'reactstrap';
import './ProductCardHeader.scss';

class CardHeader extends Component {
    render() {
        return (
            <CardTitle className={ `ProductCardTitle ProductCardTitle_${this.props.category}` }>{this.props.children}</CardTitle>
        );
    }
}

export default CardHeader;