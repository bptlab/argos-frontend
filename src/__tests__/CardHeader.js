import React from 'react';
import ProductCardHeader from '../DashboardView/ProductCardGrid/ProductCard/ProductCardHeader/ProductCardHeader.js';
import renderer from 'react-test-renderer';

test('Render DashboardView Header', () => {
    const component = renderer.create(
        <ProductCardHeader category="WARNING">Name of Product</ProductCardHeader>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});