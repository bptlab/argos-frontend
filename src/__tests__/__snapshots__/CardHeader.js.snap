exports[`test Render DashboardView Header 1`] = `
<h4
  className="card-title ProductCardTitle ProductCardTitle_WARNING">
  Name of Product
</h4>
`;
